### ISSUE TITLE
Delete a folder from the explorer pane doesn't work

### ISSUE THREAD

**Issue by jeromu** [Microsoft/vscode#10397]
- VSCode Version: 1.4.0
- OS Version: Windows Server 2008 R2

Steps to Reproduce:

1. Right-click on a **folder** that contains one or more files and folders, and select 'Delete'
2. (You'll be prompted to move to recycle bin)
3. Click 'Move to recycle bin'
4. (Notice that the folder doesn't show up in the explorer anymore)
5. Click the refresh button

Observed:
- The folder comes back, it was never moved to the recycle bin

Note: Deleting a single file works, but deleting a folder doesn't.

After looking more into it... it seems that if there is a problem deleting the folder (due to permissions), VSCode doesn't let the user know, so the problem is about reporting a problem deleting a file/folder.

**Comment by bpasero** [comment]
@jeromu I assume you are using TypeScript?

**Comment by jeromu** [comment]
No, I'm using purely all Javascript (ES6), but I do have a 'typings' directory, since that seemed to improve the auto-completion of not only typescript files, but also javascript files.

**Comment by dbaeumer** [comment]
This is still a problem with the tsserver since it is used for JS files as well. The issue in TS land is: https://github.com/Microsoft/TypeScript/issues/7762

**Comment by devuxer** [comment]
@dbaeumer,

I would like to propose reopening this one because the **core** problem is the lack of a warning that a folder is locked for deletion and the disappearance of the (not actually deleted) folder from the vscode Explorer tree, not that there is a common scenario (tsserver) that causes folders to be incorrectly locked.

I came to this issue because I noticed certain files in my build folder that weren't supposed to be there anymore. I though, well, if I delete the whole build folder and run my build process again, the files will disappear. I tried this, and it didn't work. I scoured my project looking for any reference to these ghost files, but I couldn't find any. After killing half an hour or so on this, it finally occurred to me to hit refresh after deleting the build folder. This is when I realized the build folder only *looked* like it was being deleted but actually wasn't. I then viewed the folder in Windows explorer and tried deleting it from there. This time, I got a proper warning and was able to diagnose that one of the files was open in another application. Had vscode given me the same warning and not hidden the folder, I would have saved a lot of time and aggravation.
